geniza/corpus/ja.py

```diff
@@ -80,28 +80,31 @@ def arabic_to_ja(text):
 
 
 def arabic_or_ja(text, boost=True):
-    """Convert text to arabic or judaeo-arabic string; boost arabic by default"""
-    # if there is no arabic text, return as is
-    if not contains_arabic(text):
-        return text
+    # any arabic phrase or word
+    sr = re.compile(r'"[ \u0600-\u06FF]+"|[\u0600-\u06FF]+')
```

Comment:
let's move the regex compile step outside the method (so we only compile once) and name it something meaningful